src/overlays/actors/ovl_Bg_Hidan_Sekizou/z_bg_hidan_sekizou.c

```diff
@@ -110,25 +112,323 @@ static ColliderJntSphInit D_8088E258 = {
     6,
     D_8088E180,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Sekizou/func_8088CEC0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Sekizou/BgHidanSekizou_Init.s")
+CollisionCheckInfoInit D_8088E268 = { 1, 40, 240, MASS_IMMOVABLE };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1500, ICHAIN_STOP),
+};
+
+void* sFireballsTexs[] = { gFireTempleFireball0Tex, gFireTempleFireball1Tex, gFireTempleFireball2Tex,
+                           gFireTempleFireball3Tex, gFireTempleFireball4Tex, gFireTempleFireball5Tex,
+                           gFireTempleFireball6Tex, gFireTempleFireball7Tex };
+
+void func_8088CEC0(BgHidanSekizou* this, s32 arg1, s16 arg2) {
+    s32 i;
+    s32 start = arg1 * 3;
+    s32 sp34 = start + 3;
+    f32 sp30 = Math_SinS(arg2);
+    f32 sp2C = Math_CosS(arg2);
+
+    for (i = start; i < sp34; i++) {
+        ColliderJntSphElement* element = &this->collider.elements[i];
+
+        element->dim.worldSphere.center.x = this->dyna.actor.home.pos.x + (sp2C * element->dim.modelSphere.center.x) +
+                                            (sp30 * element->dim.modelSphere.center.z);
+        element->dim.worldSphere.center.y = (s16)this->dyna.actor.home.pos.y + element->dim.modelSphere.center.y;
+        element->dim.worldSphere.center.z = this->dyna.actor.home.pos.z - (sp30 * element->dim.modelSphere.center.x) +
+                                            (sp2C * element->dim.modelSphere.center.z);
+        element->info.toucherFlags |= TOUCH_ON;
+        element->info.ocElemFlags |= OCELEM_ON;
+    }
+}
+
+void BgHidanSekizou_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgHidanSekizou* this = THIS;
+    s32 i;
+    CollisionHeader* sp40 = NULL;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DPM_UNK);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->dyna.actor, &D_8088E258, this->elements);
+    for (i = 0; i < 6; i++) {
```

Comment:
```suggestion
    for (i = 0; i < ARRAY_COUNT(this->collider.elements); i++) {
```

Reply:
honestly a define would probably work better for all of these, but that would need to happen later on.